cdap-docs/developer-manual/source/building-blocks/schedules.rst

```diff
@@ -167,37 +166,19 @@ Triggers
-- ``.triggerByPartition(String datasetName, int numPartitions)``: This is triggered by new
-  partitions in the named dataset (it must be a ``PartitionedFileSet`` or a
-  ``TimePartitionedFileSet``). While a single notification for new partitions in the
-  dataset suffices to to create a job for the schedule, it will remain in *pending trigger*
-  state until enough partitions have arrived to exceed the ``numPartitions`` parameter.
-- ``.triggerOnProgramStatus(String namespace, String application, String applicationVersion, ProgramType type, String program, ProgramStatus... programStatuses)``:
-  This is triggered when the given program in the given namespace, application,
-  and application version transitions to any one of the given program statuses.
-- ``.triggerOnProgramStatus(String namespace, String application, ProgramType type, String program, ProgramStatus... programStatuses)``:
-  This is triggered when the given program in the given namespace and application
-  transitions to any one of the given program statuses.
-- ``.triggerOnProgramStatus(String application, ProgramType type, String program, ProgramStatus... programStatuses)``:
-  This is triggered when the given program in the given application, but the same namespace
-  and application version transitions to any one of the given program statuses.
-- ``.triggerOnProgramStatus(ProgramType type, String program, ProgramStatus... programStatuses)``: This is triggered when
-  the given program in the same namespace, application, and application version
-  transitions to any one of the given program statuses.
-- ``.triggerOn(Trigger trigger)``: This is triggered when the given *Trigger* parameter is
-  satisfied. The parameter can be created by calling methods from *TriggerFactory*,
-  which can be obtained by calling ``getTriggerFactory`` method in an application extended
-  from ``AbstractApplication``. *TriggerFactory* contains methods to create all types of
-  triggers as described above, with two additional trigger types:
-
-  - ``TriggerFactory#and(Trigger... triggers)``: This is triggered when all the given
-    triggers in the parameter are satisfied.
-  - ``TriggerFactory#and(Trigger... triggers)``: This is triggered when any of the given
-    triggers in the parameter is/are satisfied.
+- ``partition``: This is triggered by new partitions in the named dataset (it must be a
+  ``PartitionedFileSet`` or a ``TimePartitionedFileSet``). While a single notification for
+  new partitions in the dataset suffices to to create a job for the schedule, it will
+  remain in *pending trigger* state until enough partitions have arrived to exceed
+  the ``numPartitions`` parameter.
+- ``program status``: This is triggered when a given program in the specific namespace,
+  application, and application version transitions to any one of the given program statuses.
+- ``and``: This is triggered when all the triggers contained in it are satisfied.
```

Comment:
`and` is not the type of trigger right but multiple triggers can be combined using `and`, `or` constructs.  Maybe instead of putting it in the list describe in separate paragraph what `and` and `or` does?

Reply:
however, in the `lifecycle.rst`, `and`/`or` are one of the trigger types that can be used as the value of "type" property in the trigger json object, and they are actually treated also as a type of trigger. `and`/`or` is a single trigger that can contain multiple triggers, which can also include other `and`/`or` triggers